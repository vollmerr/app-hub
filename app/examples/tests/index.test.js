import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';

import CodeSplitting, { StyledLink, ExamplesNav, ExamplesHome } from '../index';

describe('<CodeSplitting />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CodeSplitting />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a nav', () => {
    expect(wrapper.find(ExamplesNav).length).toEqual(1);
  });

  it('should render some routes', () => {
    expect(wrapper.find(Route).exists()).toEqual(true);
  });
});

describe('<ExamplesHome />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ExamplesHome />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<ExamplesNav />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ExamplesNav />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain styled links', () => {
    expect(wrapper.find(StyledLink).exists()).toEqual(true);
  });
});
