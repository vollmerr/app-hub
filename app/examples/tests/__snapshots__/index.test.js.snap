// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeSplitting /> should render correctly 1`] = `
<examples__ExamplesWrapper>
  <ExamplesNav />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/code-splitting"
      path="/code-splitting"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/redux"
      path="/redux"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/sagas"
      path="/sagas"
    />
  </Switch>
</examples__ExamplesWrapper>
`;

exports[`<ExamplesHome /> should render correctly 1`] = `
<Example
  desc={
    Array [
      "This project contains examples for how to use certain technologies used in react projects.",
      "Select a topic from above to begin!",
    ]
  }
  header="Examples"
/>
`;

exports[`<ExamplesNav /> should render correctly 1`] = `
<examples__NavWrapper>
  <examples__StyledLink
    key="/"
    to="/"
  >
    Home
  </examples__StyledLink>
  <examples__StyledLink
    key="/code-splitting"
    to="/code-splitting"
  >
    Code Splitting
  </examples__StyledLink>
  <examples__StyledLink
    key="/redux"
    to="/redux"
  >
    Redux
  </examples__StyledLink>
  <examples__StyledLink
    key="/sagas"
    to="/sagas"
  >
    Sagas
  </examples__StyledLink>
</examples__NavWrapper>
`;
